test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -325,8 +325,12 @@ contract LoansForkTest is LoansTestBase {
         uint userUnderlyingBeforeRoll = pair.underlying.balanceOf(user);
 
         vm.startPrank(user);
-        if (rollFee > 0) {
-            pair.cashAsset.approve(address(pair.loansContract), uint(rollFee));
+        IRolls.PreviewResults memory results =
+            pair.rollsContract.previewRoll(rollOfferId, pair.takerNFT.currentOraclePrice());
+        if (results.rollFee > 0) {
```

Comment:
now, using the preview, and because there's a price change, the approval should not be for `results.rollFee`, but for `uint(-results.toTaker)` if it (`results.toTaker`) is negative (taker pays)